src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -201,3 +207,46 @@ export async function uploadArtifactToS3(fileName: string, resp: CreateUploadUrl
     })
     getLogger().debug(`StatusCode: ${response.statusCode}`)
 }
+
+export async function patchVulnerableCode({ filePath, issues: [issue] }: AggregatedCodeScanIssue) {
+    const [suggestedFix] = issue.suggestedFixes ?? []
+    if (!suggestedFix) {
+        return
+    }
+    const patch = suggestedFix.code
+    const process = new ChildProcess('patch', [filePath])
+
+    const r = process.run()
+    await process.send(patch)
+    await process.send(eof)
+    const code = (await r).exitCode
+
+    if (code === 0) {
+        vscode.window
+            .showInformationMessage(
+                'Suggested Code fix was applied successfully. Run the security scan to validate the fix. To revert back to the original code use the undo command.',
+                {
+                    title: 'Run security scan',
+                }
+            )
+            .then(res => {
+                if (res?.title === 'Run security scan') {
+                    vscode.commands.executeCommand('aws.codeWhisperer.security.scan')
+                }
+            })
+
+        fs.stat(`${filePath}.orig`, function (_err, stat) {
+            if (stat) {
+                fsPromises.rm(`${filePath}.orig`)
+            }
+        })
+        fs.stat(`${filePath}.rej`, function (_err, stat) {
+            if (stat) {
+                fsPromises.rm(`${filePath}.rej`)
+            }
+        })
+        SecurityIssueHoverProvider.instance.removeFixedIssue(filePath, issue)
+    } else {
+        vscode.window.showErrorMessage(`Error applying suggested fix. Exit code: ${code}`)
```

Comment:
In this case, isn't there still a chance of `*.orig` or `*.rej` files being created?

Reply:
moving it out side `if..else`